### DISCUSSION TITLE
Navigation with tabs and NavigationStack

### DISCUSSION THREAD

**Discussion by maukur** [pointfreeco/swift-composable-architecture#1618] [Q&A]
Hello! Could anybody show an example of how I can make complex navigation with tabs and NavigationStack? The cases should include deep links and change a tab from reducers.

**Comment by JaapWijnen** [comment]
I would also be interested in an example using the new SwiftUI NavigationSplitView. I'm not sure what "the way" to do navigation is at the moment. (I know tools are coming!)

**Comment by stephencelis** [comment]
While we're looking to provide navigation tools on top of Apple's foundational tools, we have encountered enough bugs in vanilla SwiftUI navigation that, depending on your use case and requirements, it may be a showstopper for you. Here are the ones we've filed:

https://gist.github.com/mbrandonw/f8b94957031160336cac6898a919cbb7

You may particularly want to be aware of the bug in which `NavigationStack` deep linking is completely broken when the stack is embedded in a `TabView` tab.

We'd love to find workarounds and fixes for these bugs and bake them into TCA and/or our SwiftUI Navigation library, but there's only so much we can do without starting from scratch.

**Reply by JaapWijnen** [reply]
So your current suggestion would be to stick with using NavigationView? Will the new tools therefore also mostly focus on that?

**Reply by stephencelis** [reply]
@JaapWijnen I'm not confident in making any suggestions here :smile: I think folks will have to evaluate the framework bugs and workarounds for their particular use case. Some will want to take things into their own hands by dropping down to UIKit for nav. Others may be able to get by using custom hacks that help the SwiftUI tools better behave.

**Reply by maukur** [reply]
Thanks a lot for your answers, but I ask a little bit another. How I can bind my Reducer and NavigationPath?

**Reply by maukur** [reply]
I have one way of implementing it but I think it's not the best. 
I think static and observable router it's a not good ideas. 
https://github.com/maukur/composable-architecture-and-navigation-stack/tree/main/navigation
